src/julia.h

```diff
@@ -168,68 +187,90 @@ typedef struct _jl_llvm_functions_t {
     void *specFunctionObject;
 } jl_llvm_functions_t;
 
+// a Method descriptor
+// this is the stuff that's shared among different instantiations
+// (different environments) of a closure.
+typedef struct _jl_method_t {
+    JL_DATA_TYPE
+    jl_sym_t *name;  // for error reporting
+    struct _jl_module_t *module;
+    jl_sym_t *file;
+    int32_t line;
+
+    // array of all lambda infos with code generated from this one
+    jl_array_t *specializations;
+    union jl_typemap_t tfunc;
+
+    // the AST template (or, for isstaged, the generator thunk)
+    struct _jl_lambda_info_t *lambda_template;
+    jl_array_t *roots;  // pointers in generated code (shared to reduce memory)
+
+    // cache of specializations of this method for invoke(), i.e.
+    // cases where this method was called even though it was not necessarily
+    // the most specific for the argument types.
+    union jl_typemap_t invokes;
+
+    int32_t called;  // bit flags: whether each of the first 8 arguments is called
```

Comment:
still 8, but now stored in an Int32 type (because that's what's available to jltypes.c during initialization)